vulture_os/toolkit/api_parser/catonetworks/catonetworks.py

```
@@ -0,0 +1,154 @@
+                logger.error(f"[{__parser__}]:execute_query: FATAL ERROR retry count exceeded", extra={'frontend': str(self.frontend)})
+                raise CatonetworksAPIError("FATAL ERROR retry count exceeded")
+            try:
+                response = requests.post(
+                    url=self.URL,
+                    data=json.dumps(data).encode("ascii"),
+                    headers=headers,
+                    proxies=self.proxies,
+                    verify=self.api_parser_custom_certificate if self.api_parser_custom_certificate else self.api_parser_verify_ssl,
+                    timeout=30
+                )
+            except Exception as e:
+                logger.info(f"[{__parser__}]:execute_query: {e}", extra={'frontend': str(self.frontend)})
+                time.sleep(5)
+                retry_count += 1
+                continue
+            break
+        response.raise_for_status()
+        result = response.json()
+        if "errors" in result:
+            logger.error(f"[{__parser__}]:execute_query: Error while getting logs. reason : {response.content}", extra={'frontend': str(self.frontend)})
+            return False,result
+        return True,result
+
+    def get_logs(self):
+        query = '''{
+            eventsFeed(
+                accountIDs: [''' + self.catonetworks_account_id + ''']
+                marker: "''' + self.marker + '''"
+                filters: [''' + self.event_filter_string + "," + self.event_subfilter_string + ''']
+            ) {
+                marker
+                fetchedCount
+                accounts {
+                    id
+                    errorString
+                    records {
+                        time
+                        fieldsMap
+                    }
+                }
+            }
+        }'''
+        success, result = self.execute_query(query)
+        if success:
+            self.marker = result["data"]["eventsFeed"]["marker"]
```

- We want to raise an exception here, as the API needs to return a usable marker, there is no such anticipated behaviors where the marker could be equal to null after this instruction in my mind
- There is also probably a better way to save the marker : in a dedicated method, capable of catching  unexpected cases in a try except method and thus removing `if self.marker != "":` line 135 and 147, ensuring that there is no null marker used afterward